xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-link/plugin.js

```diff
@@ -111,6 +111,143 @@
+
+          });
+        },
+        itemTemplate: [
+          '<li data-id="{id}" class="ckeditor-autocomplete-item">',
+          '<div>',
+          '<span class="ckeditor-autocomplete-item-icon-wrapper">',
+          // We have to output both icon types but normally only one is defined and the other is hidden.
+          '<img src="{iconURL}"/>',
+          '<span class="{iconClass}"></span>',
+          '</span>',
+          '<span class="ckeditor-autocomplete-item-label">{label}</span>',
+          '</div>',
+          '<div class="ckeditor-autocomplete-item-hint">{hint}</div>',
+          '</li>'
+        ].join(''),
+        outputTemplate: function (item) {
+          // Upload a new attachment.
+          if (item.id === "_uploadAttachment") {
+            require(['jquery'], function ($) {
+
+              // Reuse attachment suggest code to show the file picker.
+              // Provides xwiki-attachments-store and xwiki-file-picker.
+              const requiredSkinExtensions = `<script src=` +
+                `'${XWiki.contextPath}/${XWiki.servletpath}` +
+                `skin/resources/uicomponents/suggest/suggestAttachments.js'` +
+                `defer='defer'></script>`;
+              $(CKEDITOR.document.$).loadRequiredSkinExtensions(requiredSkinExtensions);
+
+              require(['xwiki-attachments-store',
+                  'xwiki-file-picker'
+                ],
+                function (attachmentsStore, filePicker) {
+
+                  const convertFilesToAttachments = function (files, documentReference) {
+                    const attachments = [];
+                    for (var i = 0; i < files.length; i++) {
+                      const file = files.item(i);
+                      const attachmentReference = new XWiki.EntityReference(file.name, XWiki.EntityType.ATTACHMENT,
+                        documentReference);
+                      attachments.push(attachmentsStore.create(attachmentReference, file));
+                    }
+                    return attachments;
+                  };
+
+                  // Open the file picker
+                  filePicker.pickLocalFiles({
+                    accept: "*",
+                    multiple: false
+                  }).then(function (files) {
+                    const attachments = convertFilesToAttachments(
+                      files,
+                      editor.config.sourceDocument.documentReference
+                    );
+
+                    // Cancel the insertion when no file is picked.
+                    if (attachments.length === 0) {
+                      return;
+                    }
+
+                    const attachment = attachments[0];
+                    const attachmentURL = CKEDITOR.plugins.xwikiResource.getResourceURL({
+                            type: "attach",
+                            typed: false,
+                            reference: attachment.id
+                          }, editor);
```

Comment:
Unusual indenting. I guess you're indenting relative to the start of the constant name, but we normally indent relative to the start of the line (everywhere else).